### PR TITLE
tools/cephfs_mirror: fix snaps_per_cycle check

### PR DESCRIPTION
**Pull request by joscollin**
This check seems to be appropriate based on the type of `cephfs_mirror_max_snapshot_sync_per_cycle` and the min value.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by joscollin** [comment]
jenkins test make check arm64

File: src/tools/cephfs_mirror/PeerReplayer.cc [hunk 1]
```diff
@@ -1459,12 +1459,13 @@ int PeerReplayer::do_sync_snaps(const std::string &dir_root) {
...
     std::chrono::duration<double> duration = clock::now() - start;
     set_last_synced_stat(dir_root, it->first, it->second, duration.count());
-    if (--snaps_per_cycle == 0) {
-      break;
-    }
 
     last_snap_name = it->second;
     last_snap_id = it->first;
+
+    if (--snaps_per_cycle < 1) {
```
Comment (vshankar) [inline src/tools/cephfs_mirror/PeerReplayer.cc:1466]: I don't get it. Please explain.

Comment (joscollin) [reply src/tools/cephfs_mirror/PeerReplayer.cc:1466]: This check seems to be appropriate based on the type of `cephfs_mirror_max_snapshot_sync_per_cycle` and the min value. As `snaps_per_cycle` also decides whether the sync continues, putting it last seems better. These are the reasons I thought while creating it.

Comment (vshankar) [reply src/tools/cephfs_mirror/PeerReplayer.cc:1466]: > This check seems to be appropriate based on the type of `cephfs_mirror_max_snapshot_sync_per_cycle` 

which is uint64_t

and the min value.

min value: 1, which means there will be at least one iteration per sync cycle. I don't see how using `< 1` is fixing anything.

As `snaps_per_cycle` also decides whether the sync continues, putting it last seems better.

Sorry, I don't get how its better. Can you elaborate?

Comment (joscollin) [reply src/tools/cephfs_mirror/PeerReplayer.cc:1466]: Got it, min value is 1. Closing it.

**Pull request closed without merging**